ckan/templates/snippets/facet_list.html

```diff
@@ -7,86 +7,56 @@
-			    {% if items %}
-				<nav aria-label="{{ title }}">
-				    <ul class="list-unstyled nav nav-simple nav-facet">
-					{% for item in items %}
-					    {% set href = h.remove_url_param(name, item.name, extras=extras, alternative_url=alternative_url) if item.active else h.add_url_param(new_params={name: item.name}, extras=extras, alternative_url=alternative_url) %}
-					    {% set label = label_function(item) if label_function else item.display_name %}
-					    {% set label_truncated = label|truncate(22) if not label_function else label %}
-					    {% set count = count_label(item['count']) if count_label else ('%d' % item['count']) %}
-					    <li class="nav-item {% if item.active %} active{% endif %}">
-						<a href="{{ href }}" title="{{ label if label != label_truncated else '' }}">
-						    <span class="item-label">{{ label_truncated }}</span>
-						    <span class="hidden separator"> - </span>
-						    <span class="item-count badge">{{ count }}</span>
-
-							{% if item.active %}<span class="facet-close"> <i class="fa fa-solid fa-circle-xmark"></i></span>{% endif %}
-						</a>
-					    </li>
-					{% endfor %}
-				    </ul>
-				</nav>
-
-				<p class="module-footer">
-				    {% if h.get_param_int('_%s_limit' % name) %}
-					{% if h.has_more_facets(name, search_facets) %}
-					    <a href="{{ h.remove_url_param('_%s_limit' % name, replace=0, extras=extras, alternative_url=alternative_url) }}" class="read-more">{{ _('Show More {facet_type}').format(facet_type=title) }}</a>
-					{% endif %}
-				    {% else %}
-					<a href="{{ h.remove_url_param('_%s_limit' % name, extras=extras, alternative_url=alternative_url) }}" class="read-more">{{ _('Show Only Popular {facet_type}').format(facet_type=title) }}</a>
-				    {% endif %}
-				</p>
-			    {% else %}
-				<p class="module-content empty">{{ _('There are no {facet_type} that match this search').format(facet_type=title) }}</p>
-			    {% endif %}
-			{% endwith %}
-		    {% endblock %}
-		</section>
-	    {% endblock %}
+{% macro item_link(item, facet_name) -%}
+	{% set active = item["name"] in request.args.getlist(facet_name) %}
+	{% set url_args = request.args.copy() %}
+	{% if not active %}
+		{% do url_args.add(facet_name, item["name"]) %}
+	{% else %}
+		{% set current_items_in_url = url_args.getlist(facet_name) %}
+		{% do current_items_in_url.remove(item["name"]) %}
+		{% do url_args.setlist(facet_name, current_items_in_url) %}
+	{% endif %}
+	<a href="{{h.url_for('dataset.search', **url_args.to_dict(False))}}" class="list-group-item list-group-item-action align-middle{% if active%} active{%endif%}" {% if active%}aria-current="true"{%endif%}>
+		{{item["display_name"]|truncate(30)}}
+		<span class="badge {%if active%}bg-light text-dark{%else%}bg-primary{%endif%} rounded-pill">{{ item["count"] }}</span>
+		{% if active %}<span class="float-end"><i class="fa fa-solid fa-circle-xmark"></i></span>{% endif %}
+	</a>
+{%- endmacro %}
```

Comment:
I could also refactor this into a helper. Maybe it makes the implementation cleaner and better documented? thinking